ansible/roles/pico-shell/files/sshkey_auth.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+import json
+import pwd
+import sys
+from os import path
+
+import requests
+
+LOCAL_ROOT = "/opt/hacksports/local/"
+
+config_file = path.join(LOCAL_ROOT, "local_config.json")
+config = json.loads(open(config_file).read())
+SERVER = config["web_server"]
+LIMIT_BYPASS = config['rate_limit_bypass']
+TIMEOUT = 5
+
+
+def competition_active():
+    r = requests.get(SERVER + "/api/v1/status")
+    return json.loads(r.text)["competition_active"]
+
+
+def try_login(user):
+    r = requests.post(
+        SERVER + "/api/v1/user/login",
+        headers={
+            "user-agent": 'picoCTF Shell Server',
+            'Limit-Bypass': LIMIT_BYPASS
+        },
+        json={
+            "username": user,
+            "password": "-"
+        },
+        timeout=TIMEOUT)
+    return str(json.loads(r.text)['message'])
+
+
+def server_valid_user(user):
+    if not competition_active():
+        return False
+    result = try_login(user)
+    return result == "Incorrect password"
+
+
+def main(user):
+    if not server_valid_user(user):
```

Comment:
Just checking: if we deploy this, it seems the user `ubuntu` will not be able to use `authorized_keys` unless it is already in the (web) server?

Reply:
That's a great point

Reply:
Resolved in 6fb93cc